src/group_theory/nilpotent.lean

```diff
@@ -577,6 +577,47 @@ lemma nilpotency_class_quotient_le (H : subgroup G) [H.normal] [h : is_nilpotent
 lemma derived_le_lower_central (n : ℕ) : derived_series G n ≤ lower_central_series G n :=
 by { induction n with i ih, { simp }, { apply general_commutator_mono ih, simp } }
 
+lemma _root_.comm_group.center_eq_top {G : Type*} [comm_group G] : center G = ⊤ := begin
+  ext x,
+  suffices : x ∈ center G, by simpa,
+  exact λ y, mul_comm y x
+end
+
+/-- A group is commutative if the center is the whole group -/
+def _root_.group.comm_group_of_center_eq_top (h : center G = ⊤) : comm_group G :=
```

Comment:
Please move this as well.

Reply:
Right; did you see my comment above that I’ll move them once they have passed review (because it’s easier to edit while they are in a leaf module). But maybe that was too confusing :-)

Reply:
Yeah, that's fine.